app/services/customers/generate_checkout_url_service.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Customers
+  class GenerateCheckoutUrlService < BaseService
+    def initialize(customer:)
+      @customer = customer
+      @provider_customer = customer&.provider_customer
+
+      super
+    end
+
+    def call
+      return result.not_found_failure!(resource: 'customer') if customer.blank?
+
+      if provider_customer.blank?
+        return result.service_failure!(
+          code: 400,
```

Comment:
Code is supposed to be a human readable code here. The controller is already responsible for translating the error class into an HTTP code. 

Other question, we usually keep service_failure for internal service error. I think what we need here is either a `precondition_failure`  (HTTP 412, not implemented yet), or a `single_validation_failure`. What do you think?

Reply:
I would prefer a `422` over a `412`